core/lib/mobility/plugins/store_based_fallbacks.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'concurrent'
+require 'mobility'
+
+module Mobility
+  class << self
+    attr_reader :store_based_fallbacks
+  end
+
+  @store_based_fallbacks = Concurrent::ThreadLocalVar.new(I18n::Locale::Fallbacks.new)
+
+  module Plugins
+    module StoreBasedFallbacks
+      extend ::Mobility::Plugin
+
+      default true
+      requires :backend, include: :before
+
+      # Applies fallbacks plugin to attributes. Completely disables fallbacks
+      # on model if option is +false+.
+      included_hook do |_, backend_class|
+        unless options[:fallbacks] == false
+          backend_class.include(BackendInstanceMethods)
+        end
+      end
+
+      private
+
+      module BackendInstanceMethods
+        def read(locale, fallback: true, **kwargs)
+          return super(locale, **kwargs) if !fallback || kwargs[:locale]
+
+          locales = fallback == true ? Mobility.store_based_fallbacks.value[locale] : [locale, *fallback]
```

Comment:
There's already a `!fallback` check above so you know for sure that it's `true` here. `locales` will always be equal to:
```rb
Mobility.store_based_fallbacks.value[locale]
```

Reply:
That's a leftover from the stock fallbacks implementation, good point :)